common/src/main/java/io/pravega/example/tensorflow/FaceRecognizer.java

```diff
@@ -0,0 +1,237 @@
+                                graphBuilder.constant("size", new int[]{IMAGE_DIMENSION, IMAGE_DIMENSION})),
+                        graphBuilder.constant("scale", SCALE));
+        imageUtil = new ImageUtil();
+        log.info("FaceRecognizer: done initializing TensorFlow; duration = {} ms", System.currentTimeMillis() - t0);
+
+    }
+
+    public void warmup() throws Exception {
+        log.info("warmup: BEGIN");
+        final long t0 = System.currentTimeMillis();
+        InputStream image = FaceRecognizer.class.getResourceAsStream("/ben_afflek_input_2.jpg");
+        byte[] data = IOUtils.toByteArray(image);
+        locateFaces(data);
+        log.info("warmup: END; duration={} ms", System.currentTimeMillis() - t0);
+    }
+
+    @Override
+    public void close() {
+        session.close();
+    }
+
+    /**
+     * Extract the embeddings for the current face
+     *
+     * @param face in JPG format used to extract the embeddings
+     * @return embeddings in a float array
+     */
+    public float[] embeddFace(byte[] face) {
+        return executeGraph(face);
+    }
+
+    /**
+     * Runs pre-trained facenet model to extract the embeddings from the image
+     *
+     * @param jpegBytes byte array to run the facenet model on
+     * @return embeddings extracted by running the pre-trained model
+     */
+    private float[] executeGraph(byte[] jpegBytes) {
+        // Preprocess image (decode JPEG and resize)
+        try (final Tensor<?> jpegTensor = Tensor.create(jpegBytes)) {
+            final List<Tensor<?>> imagePreprocessorOutputs = session
+                    .runner()
+                    .feed(JPEG_BYTES_PLACEHOLDER_NAME, jpegTensor)
+                    .fetch(imagePreprocessingOutput.op().name())
+                    .run();
+            assert imagePreprocessorOutputs.size() == 1;
+
+            try (final Tensor<Float> preprocessedInputTensor = imagePreprocessorOutputs.get(0).expect(Float.class);
+                 final Tensor<Boolean> preprocessedPhaseTrainTensor = Tensor.create(false).expect(Boolean.class)) {
+                final List<Tensor<?>> detectorOutputs = session
+                        .runner()
+                        .feed("input", preprocessedInputTensor)
+                        .feed("phase_train", preprocessedPhaseTrainTensor)
+                        .fetch("embeddings")
+                        .run();
+                try (final Tensor<Float> resultTensor = detectorOutputs.get(0).expect(Float.class)) {
+                    final float[] outputTensor = new float[(int) resultTensor.shape()[1]];
+                    final FloatBuffer floatBuffer = FloatBuffer.wrap(outputTensor);
+                    resultTensor.writeTo(floatBuffer);
+                    return outputTensor;
+                }
+            }
+        }
+    }
+
+    /**
+     * @param cropBox  The location in the image to crop
+     * @param faceData The data of the image in jpeg format
```

Comment:
This comment is incorrect. faceData is of type Mat and it should be an uncompressed bitmap.